tokio/src/runtime/time/entry.rs

```diff
@@ -172,7 +172,12 @@ impl StateCell {
         let mut cur_state = self.state.load(Ordering::Relaxed);
 
         loop {
-            debug_assert!(cur_state < STATE_MIN_VALUE);
+            // improve the error message for things like
+            // https://github.com/tokio-rs/tokio/issues/3675
+            assert!(
+                cur_state < STATE_MIN_VALUE,
+                "mark_pending called when the timer entry is deregistered or in pending fire"
```

Comment:
This error message seems like it could be improved to me.

Reply:
Not a native English speaker. Would you mind to give some suggestions?

Reply:
How about this? I think it would be less confusing for people who hit this due to something like #3675.
```suggestion
                "mark_pending called on timer that has already fired"
```

Reply:
It looks good. But I still suggest adding the **direct** cause (i.e. `mark_pending` called when the timer entry is deregistered or in pending fire), since there may be other indirect causes (except for https://github.com/tokio-rs/tokio/issues/3675) which can trigger the assertion.

How about this?

```suggestion
                "mark_pending called when the timer entry is deregistered or in pending fire. The timer may have already been fired"
```

Reply:
Another option is this:
```suggestion
                "mark_pending called when the timer entry is in an invalid state"
```

Reply:
I prefer this one.